Add a volume-weighted Weibull IC for randomized local strength

bwspenc opened the pull request:
closes #10221

bwspenc committed:
Add a volume-weighted Weibull IC for randomized local strength closes #10221

This model is used for setting the local strength in mechanical failure
models to minimize mesh dependence.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/296606/) on 1e0eaf8 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/12954/site)

This comment will be updated on new commits.

jiangwen84 commented on modules/tensor_mechanics/test/tests/ics/volume_weighted_weibull/volume_weighted_weibull.i:
```diff
@@ -0,0 +1,58 @@
+[ICs]
+  [u_vww]
+    type = VolumeWeightedWeibull
+    legacy_generator = false
```
we should probably remove this. The default is false and this option will be deprecated.

bwspenc replied:
Removed it.

jiangwen84 commented on modules/tensor_mechanics/test/tests/ics/volume_weighted_weibull/volume_weighted_weibull.i:
```diff
@@ -0,0 +1,58 @@
+[]
+
+[Outputs]
+  exodus = true
```
I thought you used only CSVDiff. Do we need output exodus here?

bwspenc replied:
Agreed -- removed it.

jiangwen84 commented on modules/tensor_mechanics/src/ics/VolumeWeightedWeibull.C:
```diff
@@ -0,0 +1,48 @@
+#include "VolumeWeightedWeibull.h"
+
+#include "libmesh/point.h"
+#include "Distribution.h"
```
Did you use Distribution here?

bwspenc replied:
No.  Removed it.

jiangwen84 reviewed (commented):
Looks great. Minor comments.

bwspenc commented:
@jiangwen84 Thanks for the review.  I addressed all of your issues.

@aeslaughter If you have a chance, would you mind reviewing this and merging if it looks good to you? We discussed this a while back. Because of the volume correction that this does, it really needs to be done in its own class rather than within RandomIC. Since this is intended to be used for distributions of material strength, and is used for fracture modeling, I think it makes sense to put it in tensor_mechanics, but I'm open to putting it somewhere else if you think that would be better.
